src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -0,0 +1,1634 @@
+    default long nextLong(long bound) {
+        RandomSupport.checkBound(bound);
+
+        return RandomSupport.boundedNextLong(this, bound);
+    }
+
+    /**
+     * Returns a (pseudo)randomly chosen {@code long} value between the
+     * specified origin (inclusive) and the specified bound (exclusive).
+     *
+     * @param origin the least value that can be returned
+     * @param bound the upper bound (exclusive) for the returned value
+     *
+     * @return a (pseudo)randomly chosen {@code long} value between the
+     *         origin (inclusive) and the bound (exclusive)
+     *
+     * @throws IllegalArgumentException if {@code origin} is greater than
+     *         or equal to {@code bound}
+     *
+     * @implNote The default implementation simply calls
+     *           {@link RandomSupport#checkBound checkBound}(bound) and then
+     *           {@link RandomSupport#boundedNextLong boundedNextLong}(this, bound).
+     *
+     */
+    default long nextLong(long origin, long bound) {
+        RandomSupport.checkRange(origin, bound);
+
+        return RandomSupport.boundedNextLong(this, origin, bound);
+    }
+
+    /**
+     * Returns a {@code double} value (pseudo)randomly chosen from a Gaussian
+     * (normal) distribution whose mean is 0 and whose standard deviation is 1.
+     *
+     * @return a {@code double} value (pseudo)randomly chosen from a
+     *         Gaussian distribution
+     */
+    default double nextGaussian() {
+        return RandomSupport.computeNextGaussian(this);
+    }
+
+    /**
+     * Returns a {@code double} value (pseudo)randomly chosen from a Gaussian
+     * (normal) distribution with a mean and standard deviation specified by the
+     * arguments.
+     *
+     * @param mean the mean of the Gaussian distribution to be drawn from
+     * @param stddev the standard deviation (square root of the variance)
+     *        of the Gaussian distribution to be drawn from
+     *
+     * @return a {@code double} value (pseudo)randomly chosen from the
+     *         specified Gaussian distribution
+     *
+     * @throws IllegalArgumentException if {@code stddev} is negative
+     */
+    default double nextGaussian(double mean, double stddev) {
+        if (stddev < 0.0) throw new IllegalArgumentException("standard deviation must be non-negative");
+
+        return mean + stddev * RandomSupport.computeNextGaussian(this);
+    }
+
+    /**
+     * Returns a nonnegative {@code double} value (pseudo)randomly chosen from
+     * an exponential distribution whose mean is 1.
+     *
+     * @return a nonnegative {@code double} value (pseudo)randomly chosen from an
+     *         exponential distribution
+     */
+    default double nextExponential() {
+        return RandomSupport.computeNextExponential(this);
+    }
+
+    /**
+     * The property returned by {@link RandomGeneratorFactory#period() period}()
+     * if the period is unknown.
+     */
+    BigInteger UNKNOWN_PERIOD = BigInteger.ZERO;
```

Comment:
move those 3 values into RandomGeneratorFactory ?

Reply:
Will ponder on this one. I tend to agree with you since they are most likely to be used for filtering factories RandomGeneratorFactory querying was a later development. period() originally was a RandomGenerator query, but got moved when more queries were added. This will require a CSR and will come later.

Reply:
Now unused and removed.